### PR TITLE
jest & react testing

### PR DESCRIPTION
**Pull request by agneta1991**
# Jest and React testing
## Within this branch we have:

- Set up React Testing Library.
- Unit tests have been performed on operate.js and calculate.js files using Jest.
- Created unit tests for all React components (used Jest snapshots to test the components & used React Testing Library to simulate user interaction.)
- All tests passes.

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+describe('calculate function', () => {
+  beforeEach(() => {
+    operate.mockReset();
+  });
+
+  it('should reset calculator data when buttonName is "AC"', () => {
+    const result = calculate({ total: '2', next: '3', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+});
```
Comment (TedLivist) [inline src/logic/calculate.test.js]: - [ ] Kindly write more tests that cover all scenarios in the `calculate.js` file

File: src/components/buttonsdiv.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import ButtonDiv from './buttonsdiv';
+
+test('Test NavigationBar', () => {
+  // const onButtonClick = jest.fn();
```
Comment (TedLivist) [inline src/components/buttonsdiv.test.js]: - [ ] Kindly remove commented code.

File: src/components/calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import Calculator from './calculator';
+
+test('Test NavigationBar', () => {
+  const tree = render(
+    <BrowserRouter>
+      <Calculator />
+    </BrowserRouter>,
+  );
+  expect(tree).toMatchSnapshot();
+});
```
Comment (TedLivist) [inline src/components/calculator.test.js]: - [ ] Kindly add more tests using the React Testing Library to simulate user interactions, like clicking buttons. You can visit [this tutorial](https://www.robinwieruch.de/react-testing-library)

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @agneta1991 and @Zilola-Nazarova,

Good job so far!
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @agneta1991, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- All tests are passing ✔️ 
- Added test for all calculator logic ✔️ 
- Added tests for click events ✔️ 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**